### PR TITLE
fix: ipni issues

### PR DESCRIPTION
**Pull request by LexLuthr**
1. Switch to newer libipni and use headers to distinguish between ad and entry CID request. 
2. Use a cached reader for entry responses. IPNI will ask all the chunks in a row. Getting a reader each time is inefficient.
3. Fix Makefile issue
4. Fix IPNI provider HTTP URL string and remove unused variables

### CODE REVIEW

File: market/ipni/ipni-provider/ipni-provider.go [hunk 1]
```diff
@@ -140,17 +159,23 @@ func NewProvider(d *deps.Deps) (*Provider, error) {
 		httpServerAddresses[i] = addr
 	}
 
+	prCache := ttlcache.NewCache()
+	_ = prCache.SetTTL(time.Minute * 10)
+	prCache.SetCacheSizeLimit(MaxCachedReaders)
```
Comment (magik6k) [inline market/ipni/ipni-provider/ipni-provider.go:164]: Maybe we should have this wrapped into a single, cached, sector reader utility in Deps? It's one more layer, but this code will likely get duplicated in other retrieval spots

File: market/ipni/ipni-provider/ipni-provider.go [hunk 2]
```diff
@@ -491,59 +489,120 @@ func (p *Provider) handleGet(w http.ResponseWriter, r *http.Request) {
...
-						return
-					}
-					log.Errorf("failed to get entry %s for peer %s: %w", b.String(), providerID, err)
-					http.Error(w, "", http.StatusInternalServerError)
+
+		h := r.Header.Get(ipnisync.CidSchemaHeader)
+
+		switch h {
+		case ipnisync.CidSchemaAdvertisement:
+			ad, err := p.getAd(r.Context(), b, providerID)
```
Comment (magik6k) [inline market/ipni/ipni-provider/ipni-provider.go:497]: This switch would be a lot cleaner if p.getAd returned serialized bytes

File: market/ipni/ipni-provider/reader.go
Comment (magik6k) [inline market/ipni/ipni-provider/reader.go:1]: Isn't this the same cached reader we use for retrievals? (At least I somewhat vaguely remember writing this in boost)

**Review by magik6k — APPROVED** [review]
In general looks good, some comments about things which could be done a little more cleanly

File: market/ipni/ipni-provider/reader.go
Comment (LexLuthr) [reply market/ipni/ipni-provider/reader.go:1]: Yes. I will turn it into a dep and use it in IPNI and retrievals.

File: market/ipni/ipni-provider/ipni-provider.go (hunk 2 above)
Comment (LexLuthr) [reply market/ipni/ipni-provider/ipni-provider.go:497]: If I don't return the actual ad then I cannot use the func for getHead() as head needs to be a signed advertisement. Either I add more code and process getHead() separately or just return the advertisement like now.

Comment (magik6k) [reply market/ipni/ipni-provider/ipni-provider.go:491]: Makes sense, I guess then we could just do it as a separate `getAdBytes` function for use here and keep `getAd` for head getting as is.

Either case no strong opinion, can also keep this in the current shape, it works too.

Comment (LexLuthr) [reply market/ipni/ipni-provider/ipni-provider.go]: Add the the bytes function. It makes get() easier.

File: market/ipni/ipni-provider/ipni-provider.go [hunk 3]
```diff
@@ -331,98 +376,59 @@ func (p *Provider) getEntry(block cid.Cid, provider string) ([]byte, error) {
...
-		FROM
-			market_piece_deal mpd
-		INNER JOIN
-			sectors_meta sm
-		ON
-			mpd.sp_id = sm.sp_id
-			AND mpd.sector_num = sm.sector_num
-		WHERE piece_cid = $1`, piece)
+
+	reader, _, err := p.cpr.GetSharedPieceReader(ctx, piece)
```
Comment (magik6k) [inline market/ipni/ipni-provider/ipni-provider.go:382]: needs `defer reader.Close()`

**Review by magik6k — APPROVED** [review]
Looks good now, just needs the one-like reader close thing and then good to merge

File: market/ipni/ipni-provider/ipni-provider.go (hunk 3 above)
Comment (LexLuthr) [reply market/ipni/ipni-provider/ipni-provider.go:382]: Good catch.

Comment (LexLuthr) [reply market/ipni/ipni-provider/ipni-provider.go:382]: Done

**Review by magik6k — APPROVED** [review]

**Pull request merged**